src/metaxy/metadata_store/delta.py

```diff
@@ -0,0 +1,447 @@
+    Uses Polars/Narwhals components for metadata operations and relies on delta-rs for persistence.
+
+    Example:
+        ```py
+        store = DeltaMetadataStore(
+            "/data/metaxy/metadata",
+            storage_options={"AWS_REGION": "us-west-2"},
+        )
+
+        with store:
+            with store.allow_cross_project_writes():
+                store.write_metadata(MyFeature, metadata_df)
+        ```
+    """
+
+    _should_warn_auto_create_tables = False
+
+    def __init__(
+        self,
+        root_path: str | Path,
+        *,
+        storage_options: dict[str, Any] | None = None,
+        object_store_kwargs: dict[str, Any] | None = None,
+        fallback_stores: list[MetadataStore] | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Initialize Delta Lake metadata store.
+
+        Args:
+            root_path: Base directory or URI where feature tables are stored.
+            storage_options: Optional storage backend options passed to delta-rs.
+                Example: {"AWS_REGION": "us-west-2"}
+            object_store_kwargs: Keyword arguments forwarded to ``obstore.store.from_url``
+                when ``root_path`` is remote. Useful for passing ``config`` or
+                ``client_options`` dictionaries directly to obstore.
+            fallback_stores: Ordered list of read-only fallback stores.
+            **kwargs: Forwarded to [metaxy.metadata_store.base.MetadataStore][].
+        """
+        self.storage_options = storage_options or {}
+        self._object_store_kwargs = dict(object_store_kwargs or {})
+        self._display_root = str(root_path)
+        self._local_root_path: Path | None
+        self._root_uri: str
+        self._is_remote = False
+        self._object_store: ObjectStore | None = None
+
+        if isinstance(root_path, Path):
+            local_path = root_path.expanduser()
+            self._local_root_path = local_path
+            self._root_uri = str(local_path)
+        else:
+            root_str = str(root_path)
+            if "://" not in root_str:
+                local_path = Path(root_str).expanduser()
+                self._local_root_path = local_path
+                self._root_uri = str(local_path)
+            else:
+                parsed = urlparse(root_str)
+                if parsed.scheme.lower() == "file":
+                    local_path_str = url2pathname(parsed.path)
+                    if parsed.netloc and parsed.netloc not in {"", "localhost"}:
+                        local_path_str = f"//{parsed.netloc}{local_path_str}"
+                    local_path = Path(local_path_str).expanduser()
+                    self._local_root_path = local_path
+                    self._root_uri = str(local_path)
+                else:
+                    self._local_root_path = None
+                    sanitized = root_str.rstrip("/")
+                    self._root_uri = sanitized
+                    self._is_remote = True
```

Comment:
Can you explain what's about all this logic? that's really **a lot** of string/path manipulation with unclear purpose. 

A DeltaLake table just needs a single URI to be created. It means we only need to store the root path string and `os.path.join` it with the feature key to get a feature-specific table. This should be 1 line in `__init__` and 1 line in `get_table_path` or similar method.